google/resource_dataproc_job.go

```diff
@@ -0,0 +1,1068 @@
+					Optional:    true,
+					ForceNew:    true,
+					Elem:        &schema.Schema{Type: schema.TypeString},
+				},
+			},
+		},
+	}
+
+}
+
+var pySparkSchema = &schema.Schema{
+	Type:          schema.TypeList,
+	Optional:      true,
+	ForceNew:      true,
+	MaxItems:      1,
+	ConflictsWith: []string{"spark_config", "hadoop_config", "hive_config", "pig_config", "sparksql_config"},
+	Elem: &schema.Resource{
+		Schema: map[string]*schema.Schema{
+
+			"main_python_file_uri": {
+				Type:        schema.TypeString,
+				Description: "Required. The HCFS URI of the main Python file to use as the driver. Must be a .py file",
+				Required:    true,
+				ForceNew:    true,
+			},
+
+			"args": {
+				Type:        schema.TypeList,
+				Description: "Optional. The arguments to pass to the driver. Do not include arguments, such as --conf, that can be set as job properties, since a collision may occur that causes an incorrect job submission",
+				Optional:    true,
+				ForceNew:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+
+			"python_file_uris": {
+				Type:        schema.TypeList,
+				Description: "Optional. HCFS file URIs of Python files to pass to the PySpark framework. Supported file types: .py, .egg, and .zip",
+				Optional:    true,
+				ForceNew:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+
+			"jar_file_uris": {
+				Type:        schema.TypeList,
+				Description: "Optional. HCFS URIs of jar files to add to the CLASSPATHs of the Python driver and tasks",
+				Optional:    true,
+				ForceNew:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+
+			"file_uris": {
+				Type:        schema.TypeList,
+				Description: "Optional. HCFS URIs of files to be copied to the working directory of Python drivers and distributed tasks. Useful for naively parallel tasks",
+				Optional:    true,
+				ForceNew:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+
+			"archive_uris": {
+				Type:        schema.TypeList,
+				Description: "Optional. HCFS URIs of archives to be extracted in the working directory of .jar, .tar, .tar.gz, .tgz, and .zip",
+				Optional:    true,
+				ForceNew:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+
+			"properties": {
+				Type:        schema.TypeMap,
+				Description: "Optional. A mapping of property names to values, used to configure PySpark. Properties that conflict with values set by the Cloud Dataproc API may be overwritten. Can include properties set in /etc/spark/conf/spark-defaults.conf and classes in user code",
+				Optional:    true,
+				ForceNew:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+
+			"logging_config": loggingConfig(),
+		},
+	},
+}
+
+func flattenJobReference(r *dataproc.JobReference) []map[string]interface{} {
```

Comment:
Let's move all the functions that apply to all dataproc jobs into one area- these seem to have found themselves in the middle of the PySpark section.